src/policies/activityReport.js

```diff
@@ -161,8 +161,15 @@ export default class ActivityReport {
     return approverUserIds.includes(this.user.id);
   }
 
+  // This is a helper function to determine if the report is in a state where it can be edited
   reportHasEditableStatus() {
+    // if the report is in draft
     return this.activityReport.submissionStatus === REPORT_STATUSES.DRAFT
-      || this.activityReport.calculatedStatus === REPORT_STATUSES.NEEDS_ACTION;
+    // or if it's been marked as needs action
+      || this.activityReport.calculatedStatus === REPORT_STATUSES.NEEDS_ACTION || (
+    // or if it's submitted and the calculated status is not submitted for some reason
+      this.activityReport.submissionStatus === REPORT_STATUSES.SUBMITTED
+        && this.activityReport.calculatedStatus !== REPORT_STATUSES.SUBMITTED
```

Comment:
This is essentially damage control - the reports affected by this bug can't be edited if we don't add this little bit of business here. I can't see how it hurts anything - the report stays editable until the hooks have moved the report to "actually submitted" (see calculatedStatus) which is what we check on the frontend as well